quickwit-serve/src/lib.rs

```diff
@@ -96,13 +96,7 @@ pub async fn run_searcher(
         host_key,
         searcher_config.discovery_socket_addr()?,
     )?);
-    let rest_socket_addr = searcher_config.rest_socket_addr()?;
-    for peer_socket_addr in searcher_config
-        .peer_socket_addrs()?
-        .into_iter()
-        .filter(|peer_socket_addr| *peer_socket_addr != rest_socket_addr)
```

Comment:
I moved the check if in `Cluster::add_peer_node`.

Reply:
love it